### PR TITLE
ui: Pass down nspace and dc from Service model down to prometheus request

### PR DESCRIPTION
**Pull request by kaxcode**
Namespace is now being passed down and fix for mixup with the datacenter used.

### CODE REVIEW

File: ui/packages/consul-ui/app/components/topology-metrics/index.hbs [hunk 1]
```diff
@@ -17,9 +17,10 @@
     </div>
   {{#each @topology.Downstreams as |item|}}
     <TopologyMetrics::Card
-      @item={{item}}
+      @nspace={{@nspace}}
+      @dc={{@dc}}
       @service={{@service.Service}}
-      @dc={{@topology.Datacenter}}
+      @dc={{@dc}}
```
Comment (johncowen) [inline ui/packages/consul-ui/app/components/topology-metrics/index.hbs]: Is it my eyes or, is `dc` there twice?

File: ui/packages/consul-ui/app/components/topology-metrics/index.hbs [hunk 2]
```diff
@@ -17,9 +17,10 @@
     </div>
   {{#each @topology.Downstreams as |item|}}
     <TopologyMetrics::Card
-      @item={{item}}
+      @nspace={{@nspace}}
+      @dc={{@dc}}
```
Comment (johncowen) [inline ui/packages/consul-ui/app/components/topology-metrics/index.hbs:21]: Can upstreams or downstreams be in a completely different dc/nspace to the origin service you are viewing? If so you'll have to grab the dc/nspace from the current downstream/upstream itself and not use the one from the origin server.

File: ui/packages/consul-ui/app/routes/dc/services/show.js [hunk 3]
```diff
@@ -26,7 +26,7 @@ export default class ShowRoute extends Route {
 
   model(params, transition) {
     const dc = this.modelFor('dc').dc.Name;
-    const nspace = this.modelFor('nspace').nspace.substr(1);
+    const nspace = this.modelFor('nspace').nspace.substr(1) || 'default';
```
Comment (johncowen) [inline ui/packages/consul-ui/app/routes/dc/services/show.js:29]: Pretty sure this should default to just an empty string as it did previously.

**Review by johncowen — COMMENTED** [review]
I think we need to make sure we use the dc/nspace from the upstream/downstream itself, not the origin service. Ping me if thats not right, or if I'm not making sense here.

File: ui/packages/consul-ui/app/components/topology-metrics/index.hbs (hunk 2 above)
Comment (kaxcode) [reply ui/packages/consul-ui/app/components/topology-metrics/index.hbs:21]: Yes, that dc/nspace for each downstreams is within the @topology.Downstream and @topology.Upstreams items. I don't think I need to sent the Service dc/nspace in this case.

Comment (kaxcode) [reply ui/packages/consul-ui/app/components/topology-metrics/index.hbs:21]: oh, yes. For styling reasons I need to have the Service nspace/dc send down to each card. For example, does the downstream share the same nspace as the Service? If not, show the nspace with icon and tooltip.

File: ui/packages/consul-ui/app/components/topology-metrics/index.hbs [hunk 4]
```diff
@@ -73,9 +76,10 @@
       <p>{{dc}}</p>
     {{#each upstreams as |item|}}
       <TopologyMetrics::Card
+        @nspace={{@nspace}}
+        @dc={{@dc}}
```
Comment (johncowen) [inline ui/packages/consul-ui/app/components/topology-metrics/index.hbs:80]: Not entirely sure if these should use the dc and nspace from the upstream, I think we spoke about this and I'm guessing its not the case as yet.

**Review by johncowen — APPROVED** [review]
I think this is good for the moment, I had a query about upstream/downstream nspace and dcs, but I think we went over this and this is the correct outcome here.

**Pull request merged**